core/corehttp/gateway_handler.go

```diff
@@ -307,12 +383,31 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 		return
 	}
 
+	// Put the current Reed-Solomon directory entry to the cache
+	// if it is Reed-Solomon root directory.
+	if !isReedSolomonSubdirOrFile && dir.IsReedSolomon() {
+		dr, err := i.api.Unixfs().Get(context.Background(), resolvedPath)
```

Comment:
r.Context()

Reply:
This case is for the ReedSolomon root directory. Likewise, we need to keep the entry in the cache. 
If r.Context(), the context will be cancelled.